http/telegraf.go

```diff
@@ -189,6 +194,30 @@ type telegrafResponses struct {
 	TelegrafConfigs []*telegrafResponse `json:"configurations"`
 }
 
+func decodeGetTelegrafPluginRequest(ctx context.Context, r *http.Request) (*plugins.TelegrafPlugins, error) {
+	t := r.URL.Query().Get("type")
```

Comment:
nit: anyway we could change this func signature to just pass in the query vals or even the type? Makes the func signature more useful for inspecting code later 🤷‍♂.

You can also axe the context being passed in since its never being used.